stdlib/public/core/HashedCollections.swift.gyb

```diff
@@ -175,17 +123,29 @@ import SwiftShims
 // ---------------------------------------
 //
 // `Dictionary<K, V>` bridges to `NSDictionary` iff both `K` and `V` are
-// bridged.  Otherwise, a runtime error is raised.
+// bridged.  Otherwise, a runtime error is raised. This bridging is always
+// O(1), but may incur an allocation depending on the following conditions:
 //
-// * if both `K` and `V` are bridged verbatim, then `Dictionary<K, V>` bridges
-//   to `NSDictionary` in `O(1)`, without memory allocation.  Access to
-//   elements does not cause memory allocation.
+// * if both `K` and `V` are bridged verbatim, then `Dictionary<K, V>` is 
+//   toll-free bridged to NSDictionary by returning its 
+//   `_RawNativeDictionaryBuffer`.  However most of the actual implementation 
+//   of `NSDictionary` comes from the `_NativeDictionaryBuffer<K, V>`, which is 
+//   the type most `_RawNativeDictionaryBuffer`s have.
 //
-// * otherwise, `K` and/or `V` are unconditionally or conditionally bridged.
-//   In this case, `Dictionary<K, V>` is bridged to `NSDictionary` in `O(1)`,
-//   without memory allocation.  Complete bridging is performed when the first
-//   access to elements happens.  The bridged `NSDictionary` has a cache of
-//   pointers it returned, so that:
+// * regardless of if `K` and `V` are bridged verbatim, if the Dictionary is
+//   freshly allocated without any elements, then it contains the empty
+//   singleton `_RawNativeDictionaryBuffer` which is also toll-free bridged. 
+//   This uses the `NSDictionary` implementation provided by 
+//   `_RawNativeDictionaryBuffer`, which always yields nothing.
+//
+// * otherwise, `K` and/or `V` are unconditionally or conditionally bridged
+//   by inserting the Storage in a `_SwiftDeferredNSDictionary<K, V>`.
+//   Complete bridging is performed when the first access to elements happens. 
+//   This is done by constructing a 
+//   `_NativeDictionaryStorage<AnyObject, AnyObject>` containing a
+//   `_TypedNativeBuffer<AnyObject, AnyObject>`. Since AnyObject isn't
```

Comment:
`TypedBuffer` exists to allow a `NativeStorage<AnyObject, AnyObject>` to be created. RawBuffer can't run destructors because it doesn't have type info, and `NativeBuffer` has to have a `K: Hashable` constraint to fully implement `NSDictionary`.